test/test_json_util.py

```diff
@@ -270,6 +271,52 @@ def test_regex(self):
             json_util.dumps(Regex('.*', re.M | re.X),
                             json_options=LEGACY_JSON_OPTIONS))
 
+    def test_regex_pickling(self):
+        dbr = Regex(".?")
+        pickled_with_3 = b'\x80\x04\x959\x00\x00\x00\x00\x00\x00\x00\x8c\n' \
+                         b'bson.regex\x94\x8c\x05Regex\x94\x93\x94)\x81\x94}' \
+                         b'\x94(\x8c\x07pattern\x94\x8c\x02.?\x94\x8c\x05flags' \
+                         b'\x94K\x00ub.'
+        pickled_with_3_obj = pickle.loads(pickled_with_3)
+        for protocol in range(pickle.HIGHEST_PROTOCOL+1):
+            pkl = pickle.dumps(dbr, protocol=protocol)
+            if protocol == 4:
+                self.assertEqual(pickled_with_3, pkl)
```

Comment:
Testing that the pickle output is equivalent is too strict. We don't actually care if the pickle output differs, we only care that the un-pickled object is equivalent.

Reply:
Done.